README.md

```diff
@@ -1,2 +1,289 @@
+  - [`send-post`](#send-post)
+- [Examples](#examples)
+- [License](#license)
+- [Resources](#resources)
+
+## Installation
+
+### Using npx
+
+You can use `dp-cli` without installation by running:
+
+```bash
+npx dp-cli <command>
+```
+
+### Global Installation with npm
+
+To install `dp-cli` globally, use:
+
+```bash
+npm install -g dp-cli
+```
+
+Once installed, you can use `dp-cli` from anywhere in your terminal.
+
+## Setup
+
+Before using the CLI, you need to generate a keypair, set your authentication token, and register your ActivityPub actor.
+
+### Generate a Keypair
+
+Run the following command to generate a new RSA keypair:
+
+```bash
+dp-cli generate-keypair
+```
+
+This will generate a keypair and save it to your `.dprc` configuration file.
+
+### Set Your Authentication Token
+
+Obtain your `authToken` from your Distributed Press API administrator and set it using:
+
+```bash
+dp-cli set-auth-token
+```
+
+You will be prompted to enter your authentication token.
+
+### Register Your Actor
+
+Register your ActivityPub actor with the Social Inbox:
+
+```bash
+dp-cli register-actor
+```
+
+**Prompts:**
+
+1. **Enter your actor username:**  
+   _(e.g., "@username@yourdomain.com")_
+
+2. **Enter your actor URL:**  
+   _(e.g., "https://yourdomain.com/actor")_
+
+3. **Enter your public key ID:**  
+   _(e.g., "https://yourdomain.com/actor#main-key")_
+
+This will register your actor with the Social Inbox and save the details to your configuration.
+
+## Configuration
+
+The CLI uses a configuration file named `.dprc` to store API URLs, authentication tokens, keypairs, and actor information. The configuration file follows the format expected by the [`rc` module](https://www.npmjs.com/package/rc), which loads configuration options in a flexible way.
+
+### Configuration File Structure
+
+Your `.dprc` file should look like this:
+
+```json
+{
+  "dpApiUrl": "https://api.distributed.press/v1",
+  "socialInboxUrl": "https://social.distributed.press/v1",
+  "authToken": "<your-auth-token>",
+  "keypair": {
+    "publicKeyPem": "<your-public-key>",
+    "privateKeyPem": "<your-private-key>"
+  },
+  "actorUsername": "<your-actor-username>",
+  "actorUrl": "<your-actor-url>",
+  "publicKeyId": "<your-public-key-id>"
+}
+```
+
+- **`dpApiUrl`:** The base URL for the Distributed Press API.
+- **`socialInboxUrl`:** The base URL for the Social Inbox.
+- **`authToken`:** Your authentication token for API access.
+- **`keypair`:** Your public and private keys for secure interactions.
+- **`actorUsername`:** Your ActivityPub actor username.
+- **`actorUrl`:** The URL of your actor.
+- **`publicKeyId`:** The ID of your public key.
+
+_Ensure that your `.dprc` file is **not** committed to version control to keep your credentials secure._
+
+## Usage
+
+Once installed and configured, you can use the `dp-cli` command followed by specific commands to interact with the APIs.
+
+### Send a Post to Followers
+
+Publish a post to your followers:
```

Comment:
Does this publish a message to the DP site and register it in the outbox? Or ist it sending out an activity without an ID? Should test whether messages sent like this are actually visible on mastodon. We should explain how this works.